Sources/Build/BuildDelegate.swift

```diff
@@ -483,9 +515,15 @@ extension SwiftCompilerMessage {
         }
     }
 
-    private func generateProgressText(prefix: String, file: String) -> String {
-        // FIXME: Eliminate cwd from here.
-        let relativePath = AbsolutePath(file).relative(to: localFileSystem.currentWorkingDirectory!)
-        return "\(prefix) \(relativePath)"
+    private func reference(to file: String, buildPath: AbsolutePath) -> String {
+        let filePath = AbsolutePath(file)
```

Comment:
This method looks fairly expensive, would it be worth a fast path which checks to see if `file` is a literal suffix of `buildPath`, and does the cheap slice (returning a Substring, not a new String, for that matter) if it is?

Reply:
You can also probably compute `isUnderBuildPath` before the build even begins (and I think it won't change after the build starts?).

Reply:
I'll write the fast path. @aciidb0mb3r how would you compute `isUnderBuildPath` before the build begins? We start receiving the paths only as part of the build output.

Reply:
Nvm, I confused myself. I thought this was `isUnderPackagePath` or something.